syntax = "proto3";

package load_balancer;

service LoadBalancer {
  rpc GetServer (ClientRequest) returns (ServerResponse) {}
  rpc RegisterServer (ServerRegistration) returns (RegistrationResponse) {}
  rpc ReportLoad (LoadReport) returns (LoadReportResponse) {}
}

service Backend {
  rpc ProcessTask (TaskRequest) returns (TaskResponse) {}
}

message ClientRequest {
  string client_id = 1;
}

message ServerResponse {
  string server_address = 1;
}

message ServerRegistration {
  string server_address = 1;
}

message RegistrationResponse {
  bool success = 1;
}

message LoadReport {
  string server_address = 1;
  float cpu_load = 2;
}

message LoadReportResponse {
  bool success = 1;
}

message TaskRequest {
  string task_id = 1;
  string task_type = 2;  // SIMPLE or CPU_HEAVY
}

message TaskResponse {
  string result = 1;
}